fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -458,6 +468,309 @@ Value makePadding(int size) {
 	return pad.substr(0, size);
 }
 
+struct IRangeFileWriter {
+public:
+	virtual Future<Void> padEnd(bool final) = 0;
+
+	virtual Future<Void> writeKV(Key k, Value v) = 0;
+
+	virtual Future<Void> writeKey(Key k) = 0;
+
+	virtual Future<Void> finish() = 0;
+
+	virtual ~IRangeFileWriter() {}
+};
+
+struct SnapshotFileBackupCipherKey {
```

Comment:
`BlobCipherKey` provides an in-memory representation for an Encryption key used for encryption/decryption operations; any reason Backup code needs to define its own?

Reply:
Changed the structure of the code slightly